marimo/_runtime/runtime.py

```diff
@@ -1236,17 +1253,24 @@ def sigterm_handler(signum: int, frame: Any) -> None:
         else:
             signal.signal(signal.SIGTERM, sigterm_handler)
 
-    while True:
-        try:
-            request = control_queue.get()
-        except Exception as e:
-            # triggered on Windows when quit with Ctrl+C
-            LOGGER.debug("kernel queue.get() failed %s", e)
-            break
-        LOGGER.debug("received request %s", request)
-        if isinstance(request, StopRequest):
-            break
-        kernel.handle_message(request)
+    async def control_loop() -> None:
+        while True:
+            try:
+                request = control_queue.get()
```

Comment:
is there any benefit in moving some of the queues to async queues?

Reply:
i guess this queue is across threads so nvm

Reply:
I thought about that. But I wasn't really sure how to do it easily, bc we're still communicating across processes. Using an async queue will probably also have a slight performance hit, though probably negligible.